httpx/models.py

```diff
@@ -93,7 +94,11 @@ def __init__(
 
         # Handle IDNA domain names.
         if self.components.authority:
-            idna_authority = self.components.authority.encode("idna").decode("ascii")
+            # idna.encode raises InvalidCodepoint when encountering a colon, so split
+            # host and port in case the latter is specified and rejoin after encoding
+            host_port = self.components.authority.split(":")
+            host_port[0] = idna.encode(host_port[0], uts46=True).decode("ascii")
```

Comment:
Reference on the UTS 46 flag https://github.com/kjd/idna#compatibility-mapping-uts-46, maybe we should include it in a comment?

Reply:
Let's drop the uts46 flag in favor of strict=True and std3_rules=True per RFC 3986

Reply:
Do we want to fail on things like capitalization? I feel we don't necessarily want to be strict, but too much assistance can also be bad.

e.g. This will fail:
```
idna.core.InvalidCodepoint: Codepoint U+004B at position 1 of 'Königsgäßchen' not allowed
```

Reply:
Ended up using rfc3986.IRIReference.encode() instead and bypassing this entirely.